hswidget/dt1widget/tile_type_image.go

```diff
@@ -1,32 +1,33 @@
 package dt1widget
 
 import (
+	"github.com/ianling/giu"
+
+	"github.com/OpenDiablo2/HellSpawner/hswidget/dt1widget/tiletypeimage"
+
 	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
 )
 
-func getTileTypeImage(t d2enum.TileType) string {
-	switch t {
-	case d2enum.TileFloor:
-		return "floor.png"
-	case d2enum.TileLeftWall:
-		return "wall_west.png"
-	case d2enum.TileRightWall:
-		return "wall_north.png"
-	case d2enum.TileRightPartOfNorthCornerWall:
-		return "corner_upper_north.png"
-	case d2enum.TileLeftPartOfNorthCornerWall:
-		return "corner_upper_west.png"
-	case d2enum.TileLeftEndWall:
-		return "corner_upper_east.png"
-	case d2enum.TileRightEndWall:
-		return "corner_lower_south.png"
-	case d2enum.TileSouthCornerWall:
-		return "corner_lower_east.png"
-	case d2enum.TileLeftWallWithDoor:
-		return "door_west.png"
-	case d2enum.TileRightWallWithDoor:
-		return "door_north.png"
-	default:
-		return ""
-	}
+func drawTileTypeImage(t d2enum.TileType) giu.Widget {
+	return giu.Custom(func() {
+		canvas := giu.GetCanvas()
+		pos := giu.GetCursorScreenPos()
+		b := tiletypeimage.TileTypeImage(canvas, pos)
+		lookup := map[d2enum.TileType]func(){
+			d2enum.TileFloor:                      func() { b.Floor() },
+			d2enum.TileLeftWall:                   func() { b.Floor().WestWall(true) },
+			d2enum.TileRightWall:                  func() { b.Floor().NorthWall(true) },
+			d2enum.TileRightPartOfNorthCornerWall: func() { b.Floor().WestWall(false).NorthWall(true) },
+			d2enum.TileLeftPartOfNorthCornerWall:  func() { b.Floor().WestWall(true).NorthWall(false) },
+			d2enum.TileLeftEndWall:                func() { b.Floor().EastWall() },
+			d2enum.TileRightEndWall:               func() { b.Floor().SoathWall() },
+			d2enum.TileSouthCornerWall:            func() { b.Floor().Corner() },
+			d2enum.TileLeftWallWithDoor:           func() { b.Floor().WestDoor() },
+			d2enum.TileRightWallWithDoor:          func() { b.Floor().NorthDoor() },
+		}
+
+		if creator, ok := lookup[t]; ok {
+			creator()
+		}
+	})
```

Comment:
What's the rationale here for deviating from what I suggested?

Reply:
you can call `drawTileTypeImage` and store a result in some variable, but it is important to call `GetCursorScreenPos` exactly at the time, when you're building it.

Reply:
other words, canvas should be drawn at position, when we put this `giu.Custom`, not where we call `drawTileTypeImage`